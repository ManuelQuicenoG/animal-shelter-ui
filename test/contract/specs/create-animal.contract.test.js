import { provider } from '../config/initPact';
import { AnimalController } from '../../../controllers';

describe('Create an animal service', () => {
    beforeAll(async() => {
        await provider.setup();
    });
    describe('When someone wants to create an animal', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service is ready',
                uponReceiving: 'a request to create an animal',
                withRequest: {
                    method: 'POST',
                    path: '/animals',
                    body: {'name': 'New cat',
                    'breed': 'Anyway',
                    'gender': 'Male',
                    'isVaccinated': false,
                    'vaccines': []}
                },
                willRespondWith: {
                    status: 201
                }
            });
        });

        test('Then it should return correct information', async() => {
            const response = await AnimalController.register({'name': 'New cat',
            'breed': 'Anyway',
            'gender': 'Male',
            'isVaccinated': false,
            'vaccines': []});
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});