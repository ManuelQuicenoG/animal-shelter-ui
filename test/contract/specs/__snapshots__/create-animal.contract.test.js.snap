// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create an animal service When someone wants to create an animal Then it should return correct information 1`] = `""`;
